### PR TITLE
Add alpm_pkg_vercmp

### PR DESCRIPTION
**Pull request by ilpianista**
This adds alpm_pkg_vercmp

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -20,7 +20,7 @@ use num::FromPrimitive;
 
 use so::Symbol;
 use std::ffi::{CStr, CString};
-use std::os::raw::c_char;
+use std::os::raw::{c_char,c_int};
```
Comment (dns2utf8) [inline src/lib.rs]: please add a space after the comma.

File: src/lib.rs [hunk 2]
```diff
@@ -127,6 +127,20 @@ impl Alpm {
     })
   }
 
+  /// Compare two version strings and determine which one is newer.
+  ///
+  /// Returns a value comparable to the way strcmp works. Returns 1
```
Comment (dns2utf8) [inline src/lib.rs]: Is it possible to change this part of the description to a list or a table?

File: src/lib.rs [hunk 3]
```diff
@@ -127,6 +127,20 @@ impl Alpm {
...
+  /// if a is newer than b, 0 if a and b are the same version, or -1
+  /// if b is newer than a.
+  pub fn vercmp(&self, a: String, b: String) -> std::io::Result<i32> {
+    unsafe {
+      // int alpm_pkg_vercmp(const char *a, const char *b)
+      let pkg_vercmp: Symbol<fn(*const c_char, *const c_char) -> *const c_int> = try!( self.lib.get(b"alpm_pkg_vercmp") );
+
+      Ok(pkg_vercmp(a.as_ptr() as *const c_char, b.as_ptr() as *const c_char) as i32)
```
Comment (dns2utf8) [inline src/lib.rs]: Maybe we should wrap the result in [std::cmp::Ordering](https://doc.rust-lang.org/stable/std/cmp/enum.Ordering.html).

What do you think?

**Review by dns2utf8 — CHANGES_REQUESTED** [review]
Very cool :+1:

File: src/lib.rs (hunk 3 above)
Comment (ilpianista) [reply src/lib.rs]: Absolutely yes! I'll do it.

File: src/lib.rs (hunk 2 above)
Comment (ilpianista) [reply src/lib.rs]: Mmm, sorry I don't get what you mean here.

Comment (dns2utf8) [reply src/lib.rs]: The line wrap was a bit confusing, but when it returns `std::cmp::Ordering` it will become shorter. This will be ok.

**Comment by dns2utf8** [comment]
Can you rebase it?

**Review by dns2utf8 — APPROVED** [review]

**Comment by dns2utf8** [comment]
Thank you :smile:

**Pull request merged**